CONTRIBUTING.md

```diff
@@ -1,65 +1,155 @@
-# Contributor's Guide
+# Contributor Guide
+Welcome to `geocat-applications`! Thank you for your interest in contributing to this project! This guide
+describes how to contribute to `geocat-applications` and help us expand this reference for others
 
-## Create the docs locally
+If you have any questions, you can also reach us by email at [geocat@ucar.edu](geocat@ucar.edu)
 
-``` python
-conda env update -f environment.yml
+## How to Contribute
+
+There are many ways to contribute:
+
+- [Proofreading and reporting bugs](https://github.com/NCAR/geocat-applications/issues/new/choose) 🐞
+- [Requesting new features](https://github.com/NCAR/geocat-applications/issues/new/choose) 💡
+- Contribute a new [NCL to Python](#ncl-entry) page
+- Contribute a new [Python Entry](#python-entry) page
+
+## Development Workflow Overview
+This is the general development workflow to create a new page, edit, or
+expand on an existing page:
+
+**Setting up Repo and Local Development Environment**
+
+1. [Setup GitHub Account and Fork Repo](#creating-a-free-github-account)
+2. [Setup Environment](#Setup-Environment)
+3. [Create Branch for Changes](#Create-Branch-for-Changes)
+4. [Install Pre-Commit Hooks](#Install-Pre-Commit-Hooks)
+
+**Making Your Changes**
+
+1. [Understanding the Repository](#Understanding-the-Repository)
+2. [Types of `geocat-applications` pages](#Types-of-geocat-applications-pages)
+3. [Generate the Documentation Locally](#Generate-the-Documentation-Locally)
+
+**Contribute Code and Review**
+
+1. [Check Files Changed](#Check-Files-Changed)
+2. [Open a New Pull Request](#Open-a-New-Pull-Request)
+3. [Address Feedback](#Address-Feedback)
+4. [Delete Branch](#Delete-Branch)
+
+## Setting up Repo and Local Development Environment
```

Comment:
A good bit of this section is intro content covered by Project Pythia Foundations.  I wonder if we should point to that for folks who are new to Git / GitHub rather than detailing this here.

Reply:
It would be useful to point people to Pythia as further resources, but I think it would get confusing to be jumping back and forth between documentation when you only need a few of the steps to get this done for applications

Reply:
My main concern is that if you haven't worked with git at all or are unfamiliar with GitHub this section is probably not enough and some terminology used here may be confusing (e.g. locally, fork, repo) and if you have even a bit of familiarity with these tools this section is probably too much.  I think recommending something like the Project Pythia Foundations chapter and then keeping things a little tighter here gives people more of what they need and gets the git installation and configuration out of the way.

If you feel strongly about keeping it as is, I think we should at least adjust the language a bit and add some links to Project Pythia and/or Software Carpentry.

Reply:
I can see both sides here, but ultimately as long as we convey that we want people to fork our project to contribute, I'll be okay.